src/network/core.rs

```diff
@@ -27,7 +27,15 @@ impl Core {
             interfaces: Some(HashMap::new()),
         };
         // get bridge name
-        let bridge_name: String = network.network_interface.as_ref().unwrap().to_owned();
+        let bridge_name = match network.network_interface.as_ref() {
+            None => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    "no bridge provided".to_string(),
+                ))
+            }
+            Some(i) => (*i).clone(),
```

Comment:
LGTM. 

Just one nit and its non-blocking: changing above match to `match &network.network_interface` should let us remove this use of custom `de-referencing` at (*i)